volta-affiliate/install-validator-ubuntu-server-18.04-volta-nethermind.sh

```diff
@@ -0,0 +1,520 @@
+#!/bin/bash
+
+# Make the script exit on any error
+set -e
+set -o errexit
+DEBIAN_FRONTEND=noninteractive
+
+# Configuration Block - Docker checksums are the image Id
+export NETHERMIND_VERSION="nethermind/nethermind:1.8.98"
+NETHERMIND_CHKSUM="sha256:06cfc1e34b18eef586edd96120eea31c787859d1e8c010bb627d440353bf41cb"
+
+export NETHERMINDTELEMETRY_VERSION="1.0.0"
+NETHERMINDTELEMETRY_CHKSUM="sha256:3ef12c74dcc3bfe67efc54f30ffea8400ce83e932d8ba38a2989c5cf65008987"
+
+TELEGRAF_VERSION="1.15.2"
+TELEGRAF_CHKSUM="b539ed52df8ec63258ea97e91cc1145333b0345bca3f5863bebfca60df6f46e9  telegraf_1.15.2-1_amd64.deb"
+
+# Chain/Nethermind configuration
+export CHAINNAME="volta"
+BLOCK_GAS="8000000"
+CHAINSPEC_URL="https://raw.githubusercontent.com/energywebfoundation/ewf-chainspec/master/Volta.json"
+NLOG_CONFIG="https://raw.githubusercontent.com/NethermindEth/nethermind/master/src/Nethermind/Nethermind.Runner/NLog.config"
+
+# Make sure locales are properly set and generated
+apt-get update -y
+apt-get install locales -y
+echo "Setup locales"
+cat > /etc/locale.gen << EOF
+de_DE.UTF-8 UTF-8
+en_US.UTF-8 UTF-8
+EOF
+locale-gen
+echo -e 'LANG="en_US.UTF-8"\nLANGUAGE="en_US:en"\n' > /etc/default/locale
+source /etc/default/locale
+
+# Try to guess the current primary network interface
+NETIF="$(ip route | grep default | awk '{print $5}')"
+
+# Install system updates and required tools and dependencies
+echo "Installing dependencies"
+
+# Preparing iptables auto-save
+echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
+echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
+
+apt-get install -y curl net-tools dnsutils expect jq iptables-persistent debsums chkrootkit moreutils
+
+# Collecting information from the user
+
+# Get external IP
+if [ -f /sys/hypervisor/uuid ]; then
+  if [ `head -c 3 /sys/hypervisor/uuid` == "ec2" ]; then
+    echo Detected Amazon instance
+    EXTERNAL_IP="$(dig @resolver1.opendns.com ANY myip.opendns.com +short)"
+  else
+    EXTERNAL_IP="$(ip -o route get to 8.8.8.8 | sed -n 's/.*src \([0-9.]\+\).*/\1/p')"
+  fi
+elif [ -r /sys/devices/virtual/dmi/id/product_uuid ]; then
+  # If the file exists AND is readable by us, we can rely on it.
+  if [ `head -c 3 /sys/devices/virtual/dmi/id/product_uuid` == "EC2" ]; then
+    echo Detected Amazon instance
+    EXTERNAL_IP="$(dig @resolver1.opendns.com ANY myip.opendns.com +short)"
+  else
+    EXTERNAL_IP="$(ip -o route get to 8.8.8.8 | sed -n 's/.*src \([0-9.]\+\).*/\1/p')"
+  fi
+else
+  # verifying Amazon's signature, see here:
+  #    https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+  if $(curl -s -m 5 http://169.254.169.254/latest/dynamic/instance-identity/document | grep -q availabilityZone) ; then
```

Comment:
I tested installation script for centos, debian, rhel and it works fine.
For ubuntu it still gets private ip.